python/src/deltachat/account.py

```diff
@@ -609,10 +609,14 @@ def shutdown(self):
         self.log("remove dc_context references")
         # the dc_context_unref triggers get_next_event to return ffi.NULL
         # which in turns makes the event thread finish execution
+        self._event_thread.mark_shutdown()
         self._dc_context = None
 
-        self.log("wait for event thread to finish")
-        self._event_thread.wait()
+        # the following wait is left out for now because it can fail if
+        # a) python code holds an extra self._dc_context reference
+        # b) the core does not manage to send a NULL through get_next_event()
+        # self.log("wait for event thread to finish")
+        # self._event_thread.wait()
```

Comment:
Now that you break the loop, you should be able to safely wait for the event thread to shutdown. Otherwise you create lingering threads without knowing it.

Reply:
The loop is supposed to be broken already through the dc_context_unref
that should send a NULL to get_next_event() which breaks the loop and
finalizes the event thread. But this is in theory.

I changed this now to still wait with a short timeout
and log a warning if it didn't work.